apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -100,32 +108,192 @@ export class ChangeAction extends BaseRushAction {
+          `"${ChangeType[ChangeType.none]}" is provided to the ${this._bulkChangeBumpTypeParameter.longName} ` +
+          `patameter, the ${this._bulkChangeMessageParameter.longName} parameter may be omitted.`
+        );
+      }
+
+      const email: string | undefined = this._changeEmailParameter.value || this._detectEmail();
+      if (!email) {
+        throw new Error(
+          'Unable to detect git email and an email address wasn\'t provided on the ' +
+          `${this._changeEmailParameter.longName} paramter.`
+        );
+      }
 
-    return this._promptLoop()
-      .catch((error: Error) => {
-        throw new Error(`There was an error creating a change file: ${error.toString()}`);
+      const errors: string[] = [];
+
+      const comment: string = this._bulkChangeMessageParameter.value || '';
+      const changeType: string = this._bulkChangeBumpTypeParameter.value;
+      const changeFileData: Map<string, IChangeFile> = new Map<string, IChangeFile>();
+      for (const packageName of sortedProjectList) {
+        const allowedBumpTypes: string[] = Object.keys(this._getBumpOptions(packageName));
+        let projectChangeType: string = changeType;
+        if (allowedBumpTypes.length === 0) {
+          projectChangeType = ChangeType[ChangeType.none];
+        } else if (
+          projectChangeType !== ChangeType[ChangeType.none] &&
+          allowedBumpTypes.indexOf(projectChangeType) === -1
+        ) {
+          errors.push(`The "${projectChangeType}" change type is not allowed for package "${packageName}".`);
+        }
+
+        changeFileData.set(
+          packageName,
+          {
+            changes: [
+              {
+                comment,
+                changeType: projectChangeType,
+                packageName
+              } as unknown as IChangeInfo
+            ],
+            packageName,
+            email
+          }
+        );
+      }
+
+      if (errors.length > 0) {
+        for (const error of errors) {
+          console.error(error);
+        }
+        throw new AlreadyReportedError();
+      }
+
+      changeFileDataPromise = Promise.resolve(changeFileData);
+      allowOverwriteHandler = (filePath: string) => Promise.reject(new Error(`Changefile ${filePath} already exists`));
+    } else if (this._bulkChangeBumpTypeParameter.value || this._bulkChangeMessageParameter.value) {
+      throw new Error(
+        `The ${this._bulkChangeParameter.longName} flag must be provided with the ` +
+        `${this._bulkChangeBumpTypeParameter.longName} and ${this._bulkChangeMessageParameter.longName} parameters.`
+      );
+    } else {
+      const promptModule: inquirer.PromptModule = inquirer.createPromptModule();
+
+      this._changeComments = ChangeFiles.getChangeComments(this._getChangeFiles());
+      changeFileDataPromise = this._promptForChangeFileData(promptModule, sortedProjectList).then((changeFileData) => {
```

Comment:
There are some issues with the promises in this PR.  Let's chat offline about it.